### DISCUSSION TITLE
Can't make it work with Gitlab

### DISCUSSION THREAD

**Discussion by loranger** [outline/outline#4436] [Self Hosting]
Hello,

I struggle with gitlab integration.
I try to setup an outline using docker for my team and I want to use our (paying group owned) gitlab

I did follow [the doc](https://app.getoutline.com/s/770a97da-13e5-401e-9f8a-37949c19f97e/doc/gitlab-GjNVvyv7vW) and setup my env like this:

```
# GITLAB Auth

OIDC_CLIENT_ID=12xxxxxxxxxxx34
OIDC_CLIENT_SECRET=56xxxxxxxxxxxxxx78
OIDC_AUTH_URI=https://gitlab.com/oauth/authorize
OIDC_TOKEN_URI=https://gitlab.com/oauth/token
OIDC_USERINFO_URI=https://gitlab.com/oauth/userinfo
OIDC_USERNAME_CLAIM=username
OIDC_DISPLAY_NAME=GitLab
OIDC_SCOPES=openid email
```

I also added a [new Application in our gitlab](https://docs.gitlab.com/ee/integration/oauth_provider.html#group-owned-applications) as found out in [callback url](https://github.com/outline/outline/discussions/3450)

<img width="820" alt="SCR-20221116-hjc-2" src="https://user-images.githubusercontent.com/6014/202177047-db02b0ce-6769-4874-a592-ee950fa2642a.png">

But I can't make it work.
Whatever I use as callback url or OIDC_USERNAME_CLAIM, I always end up with a `The redirect URI included is not valid.` error.

<img width="446" alt="SCR-20221116-ihg" src="https://user-images.githubusercontent.com/6014/202178834-4dfc9612-cc87-4a8e-b349-ac78f7b5ac0f.png">

It may be a gitlab group issue, but I can't figure out what I miss.

Can you help ?

**Comment by MDXDave** [comment]
Already fixed: https://github.com/outline/outline/issues/4441
